web/src/settings_components.ts

```diff
@@ -136,6 +140,15 @@ export function get_property_value(
         return group[property_name as keyof UserGroup];
     }
 
+    if (custom_profile_field) {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        const value = custom_profile_field[property_name as keyof CustomProfileField];
+        if (property_name === "display_in_profile_summary" && value === undefined) {
+            return false;
```

Comment:
display_in_profile_summary is not set in many cases in `realm.custom_profile_fields`. Instead of messing around why it is not being set, I felt this change would be safer for now as to not introduce any unnecessary complications and refactors around it.

Reply:
Yeah, that's intended -- it's only present when true, I think.

But the API documentation doesn't explain this correctly. I think the bug is that `display_in_profile_summary` is not tagged `optional` in `zerver/openapi/zulip.yaml`. Maybe worth a thread in the "api documentation" stream.

Reply:
Followed up at https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/display_in_profile_summary.20not.20present.20in.20response/near/1789472